aspnetcore/fundamentals/minimal-apis/responses.md

```diff
@@ -0,0 +1,193 @@
+
+The `IResult` interface defines a contract that represents the result of an HTTP endpoint. The static [Results](<xref:Microsoft.AspNetCore.Http.Results>) class and the static [TypedResults](<xref:Microsoft.AspNetCore.Http.TypedResults>) are used to create varying `IResult` objects that represent different types of responses.
+
+The following example uses the built-in result types to customize the response:
+
+[!code-csharp[](7.0-samples/todo/Program.cs?name=snippet_getCustom)]
+
+### TypedResults x Results
+
+The `Results` and `TypedResults` static classes provide a similar set of results helpers, however, the `Results` static class helpers return type is `IResult`, that means that a conversion is needed when the concrete type is needed, eg.: unit testing, while `TypedResults` return type is one of the `IResult` implementation types (<xref:Microsoft.AspNetCore.Http.HttpResults> namespace).
+
+An advantage of using `TypedResults` is that the implementation type automatically includes the response type metadata for the endpoint.
+
+Consider the follow endpoint, which a `200 OK` status code with the expected JSON response is produced.
+
+```csharp
+app.MapGet("/hello", () => Results.Ok(new Message() {  Text = "Hello World!" }))
+    .Produces<Message>();
+```
+
+In order to document this endpoint correctly the extensions method `Produces` was called. However, using `TypedResults` automatically includes the metadata for the endpoint as shown in the following code.
+
+```csharp
+app.MapGet("/hello", () => TypedResults.Ok(new Message() {  Text = "Hello World!" }));
+```
+
+More information about describing a response type can be found in [OpenAPI support in minimal APIs](/aspnet/core/fundamentals/minimal-apis/openapi#describe-response-types-1))
+
+### Results<TResult1, TResultN>
+
+When using the static `TypedResult` class to create the `IResult` objects and multiple `IResult` return types are needed, returning [`Result<TResult1, TResultN>`](/dotnet/api/microsoft.aspnetcore.http.httpresults.results-2) union type from a Minimal endpoint handler is an alternative over returning `IResult` because the generic union types automatically retain the endpoint metadata and, since the `Results<TResult1, TResultN>` union types implement implicit cast operators, the compiler can automatically convert the types specified in the generic arguments to an instance of the union type. 
+
+This has the added benefit of providing compile-time checking that a route handler actually only returns the results that it declares it does. Attempting to return a type that isn’t declared as one of the generic arguments to `Results<>` results in a compilation error.
+
+Consider the follow endpoint, which a `400 BadRequest` status code is returned when the `orderId` it greater than `999` otherwise will produce a `200 OK` with the expected content.
```

Comment:
```suggestion
Consider the following endpoint, for which a `400 BadRequest` status code is returned when the `orderId` is greater than `999`. Otherwise, it produces a `200 OK` with the expected content.
```